executor/analyze.go

```diff
@@ -407,15 +429,21 @@ func (e *AnalyzeColumnsExec) buildStats() (hists []*statistics.Histogram, cms []
 			return nil, nil, err
 		}
 		sc := e.ctx.GetSessionVars().StmtCtx
+		rowCount := int64(0)
 		if e.pkInfo != nil {
-			pkHist, err = statistics.MergeHistograms(sc, pkHist, statistics.HistogramFromProto(resp.PkHist), int(e.maxNumBuckets))
+			respHist := statistics.HistogramFromProto(resp.PkHist)
+			rowCount = int64(respHist.TotalRowCount())
+			pkHist, err = statistics.MergeHistograms(sc, pkHist, respHist, int(e.maxNumBuckets))
 			if err != nil {
 				return nil, nil, err
 			}
 		}
 		for i, rc := range resp.Collectors {
-			collectors[i].MergeSampleCollector(sc, statistics.SampleCollectorFromProto(rc))
+			respSample := statistics.SampleCollectorFromProto(rc)
+			rowCount = respSample.Count + respSample.NullCount
```

Comment:
Should this be `rowCount += xxx`

Reply:
No, because they processed the same rows, and we just split the row into different collectors.